### PR TITLE
test: add `test-os.yml` for OS

### PR DESCRIPTION
**Pull request by wangliang181230**
# 1、PR 

1. 添加 `test-os.yml` 用于测试seata在各种操作系统下的运行情况。
test: add `test-os.yml` for testing the OS

2. 调整单测、actions脚本、插件版本，提高CI成功率。
Adjust unit-test, actions script and plugin versions to improve CI success rate.

3. 优化部分日志。
Optimize some logs.

# 2、测试成功日志（Test success log）：
https://github.com/wangliang181230/seata/actions?query=branch%3Atest-os-2.x

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -80,49 +80,9 @@ jobs:
       # step 4
       - name: "Test with Maven"
         run: |
-          ./mvnw -T 4C clean test -P args-for-client-test -Dspring-boot.version=${{ matrix.springboot }} -Dkotlin-maven-plugin.version=1.7.22 -Dcheckstyle.skip=true -Dlicense.skip=true -Dmaven.git-commit-id.skip=true -e -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
-
-  # job 3
-  arm64-test:
-    runs-on: ubuntu-latest
```
Comment (wangliang181230) [inline .github/workflows/test.yml:87]: This job has been moved to the `test-os.yml`

**Comment by wangliang181230** [comment]
@slievrly @funky-eyes PTAL

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -43,14 +37,14 @@ jobs:
...
-      - name: "Test with Maven"
+      - name: "Test with Maven and Java${{ matrix.java }}"
         # https://docs.github.com/cn/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#github-context
         run: |
-          if [ "${{ matrix.java }}" == "8" ]; then
-            ./mvnw -T 4C clean test -P args-for-client-test -Dspring-boot.version=${{ matrix.springboot }} -Dcheckstyle.skip=false -Dlicense.skip=false -Dmaven.git-commit-id.skip=true -e -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
-          else
-            ./mvnw -T 4C clean test -P args-for-client-test -Dspring-boot.version=${{ matrix.springboot }} -Dcheckstyle.skip=true  -Dlicense.skip=true  -Dmaven.git-commit-id.skip=true -e -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
-          fi
+          ./mvnw -T 4C clean install \
```
Comment (slievrly) [inline .github/workflows/test.yml]: Why do we need an install phase ?

Comment (wangliang181230) [reply .github/workflows/test.yml]: Oh, this is unnecessary in regular unit testing.
I only want it applied to unit-test of various OSs.
I'll fix it.

哦，这在常规测试中的确是不需要的。
我只想把它用在对各OS的测试中。
我等下修改一下。

File: .github/workflows/test-os.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
+name: "test-os"
+
+on:
+  push:
+    branches: [ test*, "*.*.*" ]
+
+jobs:
+  # job 1
```
Comment (slievrly) [inline .github/workflows/test-os.yml]: Here, it feels awkward to separate the arm64 testing from the Spring testing.

Comment (wangliang181230) [reply .github/workflows/test-os.yml]: Then I will try to merge `test.yml` and `test-os.yml`.

那我尝试把 `test.yml` 和 `test-os.yml` 合并掉。

**Review by slievrly — APPROVED** [review]
LGTM

**Pull request merged**